src/wolfio.c

```diff
@@ -438,10 +471,25 @@ int EmbedReceiveFrom(WOLFSSL *ssl, char *buf, int sz, void *ctx)
     }
 #endif /* WOLFSSL_DTLS13 */
 
-    if (!doDtlsTimeout)
-        dtls_timeout = 0;
+    do {
 
-    if (!wolfSSL_get_using_nonblock(ssl)) {
+        if (!doDtlsTimeout) {
+            dtls_timeout = 0;
+        }
+        else {
+#ifndef NO_ASN_TIME
+            if (start == 0) {
+                start = LowResTimer();
+            }
+            else {
+                dtls_timeout -= LowResTimer() - start;
+                if (dtls_timeout < 0 || dtls_timeout > DTLS_TIMEOUT_MAX)
+                    return WOLFSSL_CBIO_ERR_TIMEOUT;
```

Comment:
Is it possible for `dtls_timeout > DTLS_TIMEOUT_MAX` to be true? Can dtls_timeout increase in the loop? (Maybe true if `LowResTimer()` overflows?)

Reply:
Yes, this was an extra check for overflow, either in the clock or of the ` dtls_timeout -= LowResTimer() - start;` operation. It should be very unlikely but I believe it doesn't any harm. Should I remove it?

Reply:
It can stay in. It is a good sanity check to make sure we don't lock in this func indefinitely.